xls/modules/fpfrsqrt_32_test.cc

```diff
@@ -0,0 +1,122 @@
+// Copyright 2021 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Random-sampling test for DSLX 32-bit floating-point fast, approximate 
+// inverse sqrt.
+#include <cmath>
+#include <limits>
+
+#include "absl/random/random.h"
+#include "absl/status/status.h"
+#include "xls/common/file/get_runfile_path.h"
+#include "xls/common/init_xls.h"
+#include "xls/common/logging/logging.h"
+#include "xls/common/math_util.h"
+#include "xls/common/status/status_macros.h"
+#include "xls/ir/value_helpers.h"
+#include "xls/ir/value_view_helpers.h"
+#include "xls/modules/fpfrsqrt_32_jit_wrapper.h"
+#include "xls/tools/testbench.h"
+
+ABSL_FLAG(bool, use_opt_ir, true, "Use optimized IR.");
+ABSL_FLAG(int, num_threads, 0,
+          "Number of threads to use. Set to 0 to use all.");
+ABSL_FLAG(int64_t, num_samples, 1024 * 1024,
+          "Number of random samples to test.");
+
+namespace xls {
+
+constexpr const char kOptIrPath[] = "xls/modules/fpfrsqrt_32.opt.ir";
+constexpr const char kIrPath[] = "xls/modules/fpfrsqrt_32.ir";
+
+// Generates a float with reasonably unformly random bit patterns.
+float IndexToInput(uint64_t index) {
```

Comment:
What is the "index" arg supposed to do?

Reply:
The testbench class requires this (see xls/tools/testbench.h).